### DISCUSSION TITLE
Stream HTML page with static files and hyperlinks

### DISCUSSION THREAD

**Discussion by mateusz-glowinski** [fastapi/fastapi#6308] [Questions] [answered]
### Description

Can `FastAPI` stream HTML page with static files and hyperlinks from another service without using [RedirectResponse](https://github.com/encode/starlette/blob/master/docs/responses.md#redirectresponse)?

### Example Code

```python
from starlette.requests import Request
from starlette.responses import StreamingResponse
from starlette.background import BackgroundTask

import httpx

client = httpx.AsyncClient(base_url="http://containername:7800/")


async def _reverse_proxy(request: Request):
    url = httpx.URL(path=request.url.path, query=request.url.query.encode("utf-8"))
    rp_req = client.build_request(
        request.method, url, headers=request.headers.raw, content=await request.body()
    )
    rp_resp = await client.send(rp_req, stream=True)
    return StreamingResponse(
        rp_resp.aiter_raw(),
        status_code=rp_resp.status_code,
        headers=rp_resp.headers,
        background=BackgroundTask(rp_resp.aclose),
    )


app.add_route("/titles/{path:path}", _reverse_proxy, ["GET", "POST"])
```

### FastAPI Version

0.78.0

### Python Version

python 3.9

### Additional Context

A common architecture in micro-services is to have a gateway layer that implements a public API that passes requests to micro-services. 

```

                            =============== Docker Network =============
                            ||                                        ||
                            ||           -------> Backend Service A   ||
                            ||          /                             ||
                            ||         /                              ||
External Request ===> Gateway Service * --------> Backend Service B   ||
                            ||         \                              ||
                            ||          \                             ||
                            ||           -------> Backend Service C   ||
                            ||                                        ||
                            ============================================
```

[This solution](https://github.com/tiangolo/fastapi/issues/1788#issuecomment-1071222163) worked great for APIs.

I would go further and try use [StreamingResponse](https://github.com/encode/starlette/blob/master/docs/responses.md#streamingresponse) to display HTML page with static files and hyperlinks generated dynamically.
`RedirectResponse` will not work here, since services are not available outside of docker inner network.
Mentioned solution for API's isn't working neither, due to that static files and hyperlinks have links appropriate for inner service.

**Comment by JarroVGIT** [comment]
I would recommend to use a real reverse proxy rather than using an API framework to emulate one..

**Comment by mateusz-glowinski** [comment]
> I would recommend to use a real reverse proxy rather than using an API framework to emulate one..

@JarroVGIT Thank you for your answer.
Using reverse proxy is one option, of course.
I just wanted to try out potential solution with `FastAPI`.

**Comment by mateusz-glowinski** [comment, marked as answer]
[Here](https://stackoverflow.com/questions/73671611/how-to-stream-html-content-with-static-files-using-fastapi/73672334#73672334) is great solution.

**Comment by tiangolo** [comment]
Thanks for the help here @JarroVGIT ! 👏  🙇 

Thanks for reporting back and closing the issue @mglowinski93  👍
